src/mrinufft/trajectories/densities.py

```diff
@@ -0,0 +1,134 @@
+"""Sampling densities and methods."""
+
+import numpy as np
+import numpy.fft as nf
+import numpy.linalg as nl
+import numpy.random as nr
+import pywt as pw
+from sklearn.cluster import BisectingKMeans, KMeans
+from tqdm.auto import tqdm
+
+from .utils import KMAX
+
+
+def sample_from_density(nb_samples, density, method="random"):
+    rng = nr.default_rng()
```

Comment:
Maybe it is worth mentioning on the docstring that it will still be samples on a grid.

Reply:
Good point, I will do that